### ISSUE TITLE
Is there any way to support i18n with api comment?

### ISSUE THREAD

**Issue by darkskygit** [juhaku/utoipa#333]
Now I add api documentation to schema json through function annotation (///), I hope to provide developers with multiple language documentation of the same api, currently utoipa has any related support?

**Comment by juhaku** [comment]
Hi, unfortunately there is no support for i18n with api doc comments. This is kind of a grey area as well since there is no real specification or standard of how such feature should be implemented and provided via utoipa. I understand the need and usefulness of it but utoipa is and aims to be framework agnostic and should stay that way. 

Typically i18n features are provided by the underlying web frameworks but not necessarily. Or users could plug-in other crate which would provide this feature as they seem fit. Now in utoipa to have this implemented we do not want users to get locked to certain paradigm nor library. And we do not want users utlimately have multiple "translations" paradigms side by side in their application which only gives developers mental overhead. 

I am not against having this functionality in the library, it certainly would be cool. :gem:  Still to have it righ so that it would not lock users to a specific paradigm would need some serious design and engineering effort. 

Even better of courses if there already is a crate that is able to the translate doc comments (`///`) from a key in yaml or properties file or from somehwere else. Though I am afraid that taking in account the nature of the utoipa there is no such library that would do  it in compile time though. 

__The open issues I can think of are at least:__
* There should be a way for users to define whether they want to use _yaml_, _json_,  or _properties file_ type of translations. Or should they?, Should there be a support for some other type i18n?
* Would it be possible for users to leverage a custom crate specifically for translations? 
* How would the i18n key map to the translation within the doc comment (`///`)? Just a an empty doc comment with key simlar to `get.users.message` is not helpful for developers when editing code in editor. Should the key be last / first element in the doc comment? 
* Is there a way to define custom attributes for doc comments?
* In place translations would not be possible but translating by some key the description / title field in the generated OpenAPI spec should be doable? 
* What kind of translation support is feasible? Should there be a support for parametrisizing and plurarisizing and over all variable support?

To have the right answer there is a need for some research to see what is out there and pondering through possible solutions. However currently this is an area not given much consideration in utoipa before.

**Comment by darkskygit** [comment]
I thought utoipa might be able to provide an interface that would allow editing of the description in the apidoc at compile time or runtime?
E.g:
``` rust
let openapi_en = ApiDoc::openapi();
let openapi_cn = openapi.map(|item: ApiSchema| {
   // we can do some translate actions here to modify origin data
   // then return new data
});
router
    .route("/en.json", get(move || async { Json(openapi) }))
    .route("/cn.json", get(move || async { Json(openapi) }))
```

**Comment by juhaku** [comment]
Agree, utoipa could provide framework agnostic API for translations in which it would loop through each element needing translation and then user can freely implement the logic for translating the text. I'll add this to the todo list so this get's done at somepoint. Also PR:s are welcome :slightly_smiling_face:
